packages/server/postgres/migrations/2025-11-21T23:00:00.000Z_addOAuthProvider.ts

```diff
@@ -0,0 +1,67 @@
+import {type Kysely, sql} from 'kysely'
+
+export async function up(db: Kysely<any>): Promise<void> {
+  await db.schema
+    .createTable('OAuthAPIProvider')
+    .addColumn('id', 'text', (col) => col.primaryKey())
+    .addColumn('organizationId', 'text', (col) =>
+      col.notNull().references('Organization.id').onDelete('cascade')
+    )
+    .addColumn('name', 'text', (col) => col.notNull())
+    .addColumn('clientId', 'text', (col) => col.notNull().unique())
+    .addColumn('clientSecret', 'text', (col) => col.notNull())
+    .addColumn('redirectUris', sql`text[]`, (col) => col.notNull())
+    .addColumn('scopes', sql`text[]`, (col) => col.notNull())
+    .addColumn('createdAt', 'timestamptz', (col) => col.defaultTo(sql`now()`).notNull())
+    .addColumn('updatedAt', 'timestamptz', (col) => col.defaultTo(sql`now()`).notNull())
+    .execute()
+
+  await db.schema
+    .createTable('OAuthAPICode')
+    .addColumn('id', 'text', (col) => col.primaryKey())
+    .addColumn('clientId', 'text', (col) => col.notNull())
+    .addColumn('redirectUri', 'text', (col) => col.notNull())
+    .addColumn('userId', 'text', (col) => col.notNull())
```

Comment:
this is `varchar(100)` and it should be a foreign key & the record should cascade delete